client/v3/client.go

```diff
@@ -319,43 +266,21 @@ func (c *Client) dial(target string, creds grpccredentials.TransportCredentials,
 		defer cancel() // TODO: Is this right for cases where grpc.WithBlock() is not set on the dial options?
 	}
 
-	conn, err := grpc.DialContext(dctx, target, opts...)
+	conn, err := grpc.DialContext(dctx, c.resolver.Scheme()+":///", opts...)
```

Comment:
Can we add a comment? I don't understand why we need `:///`?

Reply:
Resolved in: https://github.com/etcd-io/etcd/pull/12706/. 

Endpoints looks like: `etcd-endpoints://{id}/#initially={list of endpoints}`

Reply:
Am I getting it correct that this endpoint/target isn't what is being used to connect to etcd server? It was purely used for gRPC to lookup the `EtcdManualResolver` for this connection at runtime and `EtcdManualResolver` will pass over the initial `Config#Endpoint` to gRPC Balancer to establish tcp connection? Theoretically, we could pass in any url that starts with `etcd-endpoints://`, e.g. `etcd-endpoints:///localhost` or `etcd-endpoints:///` (like in this PR)?

```
// Build returns itself for Resolver, because it's both a builder and a resolver.
func (r *Resolver) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
	r.CC = cc
	if r.bootstrapState != nil {
		r.UpdateState(*r.bootstrapState)
	}
	return r, nil
}
```
https://github.com/grpc/grpc-go/blob/0e8f1cda01321329950add4362b0fb0180f58b28/resolver/manual/manual.go#L56

```
func (r EtcdManualResolver) updateState() {
	if r.CC != nil {
		addresses := make([]resolver.Address, len(r.endpoints))
		for i, ep := range r.endpoints {
			addr, serverName := endpoint.Interpret(ep)
			addresses[i] = resolver.Address{Addr: addr, ServerName: serverName}
		}
		state := resolver.State{
			Addresses:     addresses,
			ServiceConfig: r.serviceConfig,
		}
		r.UpdateState(state)
	}
}
```
https://github.com/etcd-io/etcd/blob/4a1c24556c22f55e75cb5167d555a8c367ad4d2b/client/v3/internal/resolver/resolver.go#L61

Reply:
That's correct.  Bug it seems that grpc infrastructure requires the endpoint format to be: 

`schema://authority/path`

'authority' and/or 'path' can be empty. This leads to `etcd-endpoints:///` minimal name. 

`etcd-endpoints://` does not work. 
I decided to use `etcd-endpoints://{id}/#initially={list of endpoints}` for debuggability reasons, as this string gets printed in grpc logs and its good to know what we are really trying to connect to.

Reply:
Appreciate the details, @ptabor!